public/app/plugins/panel/barchart/migrations.ts

```diff
@@ -1,27 +1,51 @@
-import { PanelTypeChangedHandler } from '@grafana/data';
+import { FieldMatcherID, PanelTypeChangedHandler } from '@grafana/data';
+import { AxisPlacement } from '@grafana/ui';
 
 /*
  * This is called when the panel changes from another panel
  */
-export const changeToBarChartPanelMigrationHandler: PanelTypeChangedHandler = (
-  panel,
-  prevPluginId,
-  prevOptions,
-  prevFieldConfig
-) => {
+export const changeToBarChartPanelMigrationHandler: PanelTypeChangedHandler = (panel, prevPluginId, prevOptions) => {
   if (prevPluginId === 'graph') {
     const graphOptions: GraphOptions = prevOptions.angular;
 
+    const fieldConfig = panel.fieldConfig ?? { defaults: {}, overrides: [] };
+
     if (graphOptions.xaxis?.mode === 'series') {
-      const tranformations = panel.transformations || [];
-      tranformations.push({
-        id: 'reduce',
-        options: {
-          reducers: graphOptions.xaxis?.values ?? ['sum'],
+      const transformations = panel.transformations || [];
+      transformations.push(
+        {
+          id: 'reduce',
+          options: {
+            reducers: graphOptions.xaxis?.values ?? ['sum'],
+          },
+        },
+        {
+          id: 'transpose',
+          options: {},
+        }
+      );
+
+      panel.transformations = transformations;
+
+      // temporary, until we have a bar chart with per bar labels
+      fieldConfig.overrides.push({
+        matcher: {
+          id: FieldMatcherID.byName,
+          options: 'Name',
```

Comment:
i don't think this one is working.

![image](https://github.com/user-attachments/assets/32c30607-e51d-4b49-9b63-14913d73dc20)

maybe it should be `'Field'`?
 
![image](https://github.com/user-attachments/assets/0f235e37-3a59-4c73-954a-0c16613ef661)